File: Leetcode/350/350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {

        auto freqCnt = [](const vector<int> &a) -> unordered_map<int, int> {
            unordered_map<int, int> res;
            for (const auto &i : a) {
                ++res[i];
            }
            return res;
        };

        unordered_map<int, int> nums1_map = freqCnt(nums1);
        unordered_map<int, int> nums2_map = freqCnt(nums2);
        vector<int> res;
        int maxFreq;
        for (const auto &it : nums1_map) {
            if (nums2_map.find(it.first) != nums2_map.end()) {
                maxFreq = min(it.second, nums2_map[it.first]);
                for (int i = 0; i < maxFreq; ++i) {
                    res.push_back(it.first);
                }
            }
        }
        return res;
    }
};

File: Leetcode/350/350.go
func intersect(nums1 []int, nums2 []int) []int {
	map1 := make(map[int]int)
	for i := 0; i < len(nums1); i++ {
		map1[nums1[i]]++
	}
	map2 := make(map[int]int)
	for i := 0; i < len(nums2); i++ {
		map2[nums2[i]]++
	}
	var ans []int
	for k, val1 := range map1 {
		val2, ok := map2[k]
		if ok {
			if val1 > val2 {
				for i := 0; i < val2; i++ {
					ans = append(ans, k)
				}
			} else {
				for i := 0; i < val1; i++ {
					ans = append(ans, k)
				}
			}
		}
	}
	return ans
}